test/integration/lib/philipsHue_server.js

```diff
@@ -160,9 +102,112 @@ var philipsHue_server = (function() {
      });
    });
   }
+   
+  // Callback methods
+
+  _server.use(function (req, res, next) { 
+    // foxbox does not have the json header, so need to 
+    // inject it here so the message body can be properly captured
+    req.headers['content-type'] = 'application/json';
+    last_received_cmd = req.method + ':' + req.originalUrl;
+    next();
+  });
+
+  _server.get('/', function (req, res) {
+    res.status(200).sendFile(
+      path.join(__dirname,'/html/philips_initial.html'));  
+  });
+
+  _server.get('/api/:foxbox/', function (req, res) {   
+    
+     if (foxboxId === undefined) {
+      if (auth_status === false) {
+        foxboxId = req.params.foxbox;
+        res.sendStatus(200);  
+      }
+      else if (auth_status === true) {
+        res.status(200).send([{
+              'error': {
+                  'type': 1,
+                  'address': '/',
+                  'description': 'unauthorized user'
+              }
+          }]);
+      }     
+    }
+    else {
+      res.status(200).json(api_resp);  
+    }     
+  });
+
+  _server.post('/api', bodyParser.json(),function (req, res) {
+    
+    if (foxboxId === undefined) {
+        res.status(200).send([{
+            'error': {
+                'type': 101,
+                'address': '',
+                'description': 'link button not pressed'
+            }
+        }]);
+      }
+      else {
+        if (req.body.devicetype === 'foxbox_hub') {
+          res.status(200).send([{
+            'success': {
+            'username': 'simulatedPhilipsHub'
+            }
+          }]);
+        }
+      }
+  });
+
+  _server.get('/api/:foxboxId/lights', function (req, res) {
+    if (req.params.foxboxId === foxboxId){ 
+      res.status(200).json(light_status); 
+    }
+    else {
+      res.status(404); 
```

Comment:
but for each case, the response contents are slightly different.  since there are only 2 places that does this check, it think it is more readable unless we are expanding the checks to other methods.